### PR TITLE
Youtube video downloader in rust

### LINKED ISSUE #930: YouTube downloader in rust
# Description

Proposal to build a simple YouTube video downloader in rust.

## Language

Rust

## Checklist:

- **Details on the Script**

| name            | about                                                  | labels  | assignees                                                      |
| --------------- | ------------------------------------------------------ | ------- | -------------------------------------------------------------- |
| ytdl | Download videos from YouTube from the commandline | GSSOC21 | @gokulsoumya |

**Additional context**: 

#811, #838

### PR DESCRIPTION
**Pull request by gokulsoumya**
# Description

This rust program is a rudimentary youtube video downloader.

Fixes #930

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works

ytdl demo

### CODE REVIEW

File: Rust/Youtube_Downloader/.gitignore [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# will have compiled files and executables
+/target/
+
+.*.sw[opn]
+
+test.log
+
+/build/
+
+*.mp[34]
```
Comment (HarshCasper) [inline Rust/Youtube_Downloader/.gitignore:11]: Put this on the `.gitignore` present in the root directory

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Tiny change. Good to merge :tada:

File: Rust/Youtube_Downloader/.gitignore (hunk 1 above)
Comment (gokulsoumya) [reply Rust/Youtube_Downloader/.gitignore]: Resolved and pushed 👍🏾

**Comment by gokulsoumya** [comment]
Made the change 👍🏾

**Pull request merged**